src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/keystone/KeystoneV3IdentityRequestHelper.java

```diff
@@ -0,0 +1,110 @@
+package org.fogbowcloud.saps.engine.core.dispatcher.email.keystone;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.commons.codec.Charsets;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.log4j.Logger;
+import org.eclipse.jetty.http.HttpStatus;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class KeystoneV3IdentityRequestHelper {
+
+    public static final String PROJECT_ID = "projectId";
+    public static final String PASSWORD = "password";
+    public static final String USER_ID = "userId";
+    public static final String AUTH_URL = "authUrl";
+    private static final Logger LOGGER = Logger.getLogger(KeystoneV3IdentityRequestHelper.class);
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final String ACCEPT = "Accept";
+    private static final String JSON_CONTENT_TYPE = "application/json";
+    private static final String X_SUBJECT_TOKEN = "X-Subject-Token";
+    private static final String IDENTITY_PROP = "identity";
+    private static final String PROJECT_PROP = "project";
+    private static final String METHODS_PROP = "methods";
+    private static final String SCOPE_PROP = "scope";
+    private static final String AUTH_PROP = "auth";
+    private static final String USER_PROP = "user";
+    private static final String ID_PROP = "id";
+    private static final String V3_TOKENS_ENDPOINT_PATH = "/v3/auth/tokens";
+
+    public static String createAccessId(Map<String, String> credentials) throws Exception {
```

Comment:
As commented [here](https://github.com/ufcg-lsd/saps-engine/pull/88#discussion_r386968642), return KeystoneException class and use it again.

Reply:
I replaced the Keystone Exception with more specific exceptions like JSONException and IOException.